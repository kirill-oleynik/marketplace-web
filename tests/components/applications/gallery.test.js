const React = require('react');
const { shallow } = require('enzyme');
const toJSON = require('enzyme-to-json').default;
const ApplicationGallery = require('../../../app/components/applications/gallery').default;

const renderGallery = () => (
  <ApplicationGallery
    slides={[
      {
        id: 1,
        title: 'Hello World!',
        description: 'Pam Pam',
        url: 'https://example.com/1.png'
      },
      {
        id: 2,
        title: 'Bye World!',
        description: 'Hey Ho',
        url: 'https://example.com/2.png'
      }
    ]}
  />
);

describe('#render', () => {
  test('it renders correctly', () => {
    const component = shallow(
      renderGallery()
    );

    expect(
      toJSON(component)
    ).toMatchSnapshot();
  });
});

describe('#handleSlideChange', () => {
  test('it changes currentIndex state', () => {
    const component = shallow(
      renderGallery()
    );

    expect(component.state().currentIndex).toEqual(0);
    component.instance().handleSlideChange(1);
    expect(component.state().currentIndex).toEqual(1);
  });
});

describe('#handleFullscreenClick', () => {
  test('it changes isOpen state', () => {
    const component = shallow(
      renderGallery()
    );

    expect(component.state().isOpen).toEqual(false);
    component.instance().handleFullscreenClick();
    expect(component.state().isOpen).toEqual(true);
  });
});

describe('#handleCloseClick', () => {
  test('it changes isOpen state', () => {
    const component = shallow(
      renderGallery()
    );

    expect(component.state().isOpen).toEqual(false);
    component.instance().handleFullscreenClick();
    expect(component.state().isOpen).toEqual(true);
    component.instance().handleCloseClick();
    expect(component.state().isOpen).toEqual(false);
  });
});
