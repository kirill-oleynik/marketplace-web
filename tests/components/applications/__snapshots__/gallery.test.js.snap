// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render it renders correctly 1`] = `
<div>
  <ImageGallery
    autoPlay={false}
    disableArrowKeys={false}
    disableSwipe={false}
    disableThumbnailScroll={false}
    indexSeparator=" / "
    infinite={true}
    items={
      Array [
        Object {
          "original": "https://example.com/1.png",
          "thumbnail": "https://example.com/1.png",
        },
        Object {
          "original": "https://example.com/2.png",
          "thumbnail": "https://example.com/2.png",
        },
      ]
    }
    lazyLoad={false}
    onSlide={[Function]}
    renderFullscreenButton={[Function]}
    renderLeftNav={[Function]}
    renderPlayPauseButton={[Function]}
    renderRightNav={[Function]}
    renderThumbnailsLeftNav={[Function]}
    renderThumbnailsRightNav={[Function]}
    showBullets={false}
    showFullscreenButton={true}
    showIndex={false}
    showNav={true}
    showPlayButton={false}
    showThumbnails={true}
    showThumbnailsNav={true}
    slideDuration={450}
    slideInterval={2000}
    slideOnThumbnailHover={false}
    startIndex={0}
    swipingTransitionDuration={0}
    thumbnailPosition="bottom"
    useBrowserFullscreen={true}
  />
  <ApplicationGalleryModal
    currentIndex={0}
    isOpen={false}
    onCloseClick={[Function]}
    renderLeftNav={[Function]}
    renderRightNav={[Function]}
    slide={
      Object {
        "description": "Pam Pam",
        "id": 1,
        "title": "Hello World!",
        "url": "https://example.com/1.png",
      }
    }
    slidesLength={2}
  />
</div>
`;
